### ISSUE TITLE
Suggestion: Add readonly Option for Subjects in `rxjs-x/no-exposed-subjects`

### ISSUE THREAD

**Issue by Dmytro27Ind** [JasonWeinzierl/eslint-plugin-rxjs-x#82]
### Summary
Extend the functionality of the `rxjs-x/no-exposed-subjects` rule to enforce that all `Subject` instances are marked as `readonly`, with an optional configuration to enable or disable this behavior.

### Motivation
While the current rule encourages encapsulation by disallowing `public` or `protected` Subject instances, it does not enforce immutability at the class level for `private` subjects. Marking subjects as `readonly` prevents accidental reassignment of the subject instance, further enhancing code safety and clarity. This aligns with RxJS best practices by ensuring that Subject properties remain consistent throughout the lifecycle of the object.

Proposed ### Option
Introduce a new configuration option:

```js
{
  "rxjs-x/no-exposed-subjects": [
    "error",
    { "requireReadonly": true }
  ]
}
```

### Incorrect Code:

```ts
import { Subject } from "rxjs";

class Answers {
  private _answers: Subject<number>; // Missing readonly

  get answers() {
    return this._answers.asObservable();
  }
}
```

### Correct Code:

```ts
import { Subject } from "rxjs";

class Answers {
  private readonly _answers: Subject<number>; // Readonly enforced

  get answers() {
    return this._answers.asObservable();
  }
}
```

**Comment by JasonWeinzierl** [comment]
There's a rule `@typescript-eslint/prefer-readonly` that would handle this for Subjects and more. Does that suffice instead of adding it to this rule?

**Comment by Dmytro27Ind** [comment]
Thank you for the response!

While `@typescript-eslint/prefer-readonly` is a valuable rule for ensuring immutability across a codebase, there are some reasons why adding a readonly check specifically for `Subject` instances within `rxjs-x/no-exposed-subjects` could still be beneficial:

### Context-Specific Enforcement:

- `@typescript-eslint/prefer-readonly` applies broadly, across all properties that can be readonly, which may lead to noise or unwanted recommendations in contexts where immutability isn't critical.
- By focusing exclusively on `Subject` properties, `rxjs-x/no-exposed-subjects` keeps the enforcement tightly scoped to RxJS usage, aligning directly with the rule's purpose.

**Comment by JasonWeinzierl** [comment]
Got it.  I would likely implement this as a separate rule `rxjs/prefer-readonly-subjects`, also added to the `strict` configuration.

Also could you point me to the "RxJS best practices" documentation or article you mentioned in the description?  I can link that as Further Reading in the rule's documentation once it's implemented.

**Comment by JasonWeinzierl** [comment]
While I agree with the proposal, please do not use LLMs to generate comments on this repository.  None of those links answer my request for best practice documentation on readonly Subjects, and I'm now suspecting that I've been talking to an AI.  Your request is heard; thank you.
